### PR TITLE
Dronin backport

### PR DESCRIPTION
**Pull request by peabody124**
This is a back port of some smaller fixes from https://github.com/d-ronin/dRonin/

These are all merged to their repository and in the latest release so should be fairly
stable. Each commit is also fairly atomic but I'd prefer not to create a PR for each one.
Reading the commit log should give a pretty clear idea what was done.

I specifically avoided PRs that touch on packaging / branding (duh). Also I avoided some
ones where there seemed to be instability and reversions. A few things that touch on more
critical systems or might be controversial, I created standalone PRs. Things using the new
DR event system or that were bug fixes related to the DR event system I also avoided.

Here are the links to the original PRs:
https://github.com/d-ronin/dRonin/pull/96
https://github.com/d-ronin/dRonin/pull/120
https://github.com/d-ronin/dRonin/pull/127
https://github.com/d-ronin/dRonin/pull/132
https://github.com/d-ronin/dRonin/pull/137
https://github.com/d-ronin/dRonin/pull/140
https://github.com/d-ronin/dRonin/pull/141
https://github.com/d-ronin/dRonin/pull/146
https://github.com/d-ronin/dRonin/pull/153
https://github.com/d-ronin/dRonin/pull/173
https://github.com/d-ronin/dRonin/pull/192
https://github.com/d-ronin/dRonin/pull/201
https://github.com/d-ronin/dRonin/pull/203/
https://github.com/d-ronin/dRonin/pull/250
https://github.com/d-ronin/dRonin/pull/251
https://github.com/d-ronin/dRonin/pull/259
https://github.com/d-ronin/dRonin/pull/262
https://github.com/d-ronin/dRonin/pull/267
https://github.com/d-ronin/dRonin/pull/275
https://github.com/d-ronin/dRonin/pull/285
https://github.com/d-ronin/dRonin/pull/286
https://github.com/d-ronin/dRonin/pull/311
https://github.com/d-ronin/dRonin/pull/340
https://github.com/d-ronin/dRonin/pull/341
https://github.com/d-ronin/dRonin/pull/353
https://github.com/d-ronin/dRonin/pull/365
https://github.com/d-ronin/dRonin/pull/378
https://github.com/d-ronin/dRonin/pull/385
https://github.com/d-ronin/dRonin/pull/413
https://github.com/d-ronin/dRonin/pull/414
https://github.com/d-ronin/dRonin/pull/426
https://github.com/d-ronin/dRonin/pull/441
https://github.com/d-ronin/dRonin/pull/444
https://github.com/d-ronin/dRonin/pull/446
https://github.com/d-ronin/dRonin/pull/453
https://github.com/d-ronin/dRonin/pull/456
https://github.com/d-ronin/dRonin/pull/465
https://github.com/d-ronin/dRonin/pull/474
https://github.com/d-ronin/dRonin/pull/484
https://github.com/d-ronin/dRonin/pull/489
https://github.com/d-ronin/dRonin/pull/490
https://github.com/d-ronin/dRonin/pull/497
https://github.com/d-ronin/dRonin/pull/498
https://github.com/d-ronin/dRonin/pull/499
https://github.com/d-ronin/dRonin/pull/516

### CODE REVIEW

**Comment by peabody124** [comment]
BTW @kubark42 if there are any particular ones you'd like me to pull out to
make back porting to cyphys easier, please let me know.

**Comment by fujin** [comment]
:-1: this is pretty shitty

**Comment by mlyle** [comment]
It's quite fine.  Just note the problem of attribution-- all updated files that come from dRonin need the dRonin copyright statement applied.

**Comment by peabody124** [comment]
@mlyle I'll wait until review and see which ones stick and them update accordingly.

@fujin this is pretty standard in OSS. keep in mind DR is building on 5+ years of our work ;-)

**Comment by kubark42** [comment]
@peabody124 Thanks! 

What do you see as the disadvantage to doing individual PRs? While I agree that it's possible to do it monolithically, I think I'll have an easier time if we do the process atomically.

P.S. Opened #2134 to help me sort through it all. I'm happy to grab those one by one and open up PRs that you can directly merge.

P.S.S. Maybe we can do a hybrid approach. There's a bunch of Sparky related stuff, for instance, that could come over as one big PR. I've labeled it as such in #2134.

**Comment by peabody124** [comment]
@kubark42 I'm ok with a bit of grouping. However, mapping 1:1 PRs is just overly labor intensive and I know would decrease my personal motivation to pull things back across. I did try and make this mostly the "no brainers" that would either dropping unused code, bug fixes, etc. Ok

**Comment by mlyle** [comment]
Seriously, if you're going to import the majority of what dRonin is doing, why don't we save the difficulty and let's find a way to work on the same project as equals?  This is very wasteful, and I'm concerned that picking things up piecemeal is going to leave regressions we've fixed or cause unexpected behavior.  It would make it easier on all of our users, too.

**Comment by kubark42** [comment]
@mlyle I like the idea in principle, but in practice I see some obstacles. You forked the project because you had a different vision for the codebase. How do you recommend reconciling that?

@peabody124 I'll do the work. It's worth it to me to have the cleaner commit structure. I've really come to respect how powerful the git history is when carefully curated so I'm happy to split your PR into subsections and then merge them in piecemeal.

**Comment by mlyle** [comment]
@kubark42 We forked because we wanted to make more rapid progress and wanted to have an equal say in project decisions.  Since you're sucking in most of our work it's obvious the vision isn't all too different ;)

**Comment by peabody124** [comment]
@kubark42 if you want to do the work I'll click the individual merge buttons :) that being said, from a technical perspective for PRs that are single commits I don't see anything that is easier with git from having a PR that is merged versus having that commit cherry-picked. In fact, reverting a cherry-pick is a simpler thing that reverting a merge.

Edit: also from my perspective (using mostly CLI tools) when you do `git log` and every other
commit is a PR merge, you see half as much information per screen about what happened.

**Comment by kubark42** [comment]
@mlyle I love working together. So what's the proposal? 

@peabody124 Ah... I'm using gitk. I guess where I stumble with `git log` is that the flat style doesn't graphically show what is an atomic feature. Because of the fact that `git log` struggles with merges, it seems to show a view which isn't terribly well aligned with the extreme "branchiness" of gitflow.
screen shot 2016-01-26 at 3 51 15 pm

I understand this is a personal preference, but I find the tree view from gitk much easier to parse:
screen shot 2016-01-26 at 3 52 09 pm

It allows me to jump around more easily and distinguish between larger features and individual merges. This level of granularity is really nice when trying to backport from OpenPilot, LibrePilot, and dRonin.

I'll see if I can do more backporting in a hybrid fashion like I did with the Sparky commits, #2145. This should keep the granularity to separate features from bugfixes, while making `git log` more useful to you.

**Comment by peabody124** [comment]
Cool. And it certainly isnt a huge deal. Normally i look at a filtered variant of git log anyway. Dont actually have gitk on my machine :)

**Comment by peabody124** [comment]
Rebased on next to make it easier to see the commits remaining for back port.

**Comment by mlyle** [comment]
@kubark42 -- come along to dRonin, join our governance, be committers seems easiest.

**Comment by mlyle** [comment]
(We have good branding, could inherit the Tau forums, collapse down the two sets of wikis, make transition easier for remaining users).

File: flight/targets/simulation/fw/pios_config.h [hunk 1]
```diff
@@ -29,112 +29,58 @@
  */
 
 
-#ifndef PIOS_CONFIG_H
-#define PIOS_CONFIG_H
+#ifndef PIOS_CONFIG_POSIX_H
```
Comment (kubark42) [inline flight/targets/simulation/fw/pios_config.h]: This seems like a mistake. The header guard should follow our pattern.

File: flight/targets/simulation/fw/pios_config.h [hunk 2]
```diff
@@ -5,10 +5,10 @@
  * @addtogroup Revolution OpenPilot Revolution support files
  * @{
  *
- * @file       pios_config.h 
+ * @file       pios_config_sim.h
```
Comment (kubark42) [inline flight/targets/simulation/fw/pios_config.h]: This is wrong. The filename is still pios_config.h.

**Comment by kubark42** [comment]
Hey, this is finally getting down small enough that GitHub isn't complaining about too many lines anymore! Good stuff.

**Comment by kubark42** [comment]
@peabody124 You mention https://github.com/d-ronin/dRonin/pull/474 in your original list, but I don't see the commit in the list, and the PR doesn't have any changes to flight/Modules/ManualControl/transmitter_control.c. Thoughts?

**Comment by kubark42** [comment]
@peabody124 Do you think we caught everything here? Can it close?

**Pull request closed without merging**